pkg/apply/fieldmanager.go

```diff
@@ -29,7 +33,7 @@ func mergeManager(ctx context.Context, clusterClient client.ClusterClient, us *u
 			continue
 		}
 		if m.Operation == metav1.ManagedFieldsOperationUpdate {
-			patch, err = csaupgrade.UpgradeManagedFieldsPatch(us, map[string]sets.Empty{fromManager: {}}, intoManager)
+			patch, err = upgradeManagedFieldsPatchLocal(us, map[string]struct{}{fromManager: {}}, intoManager)
```

Comment:
nit: 
I think that if you just set up a csaupgrade sub-package here that is the exact same thing that you are copying with a comment where you copied from would work better. I thiunk the comment at least is nice to have.

Reply:
ok, I moved it all to a new local sub-module (csaupgrade.go) in ./pkg/apply/ I added some comments to explain at the top of the file. And the commit message is there too for anyone in the future trying to understand what's going on.